assembly/src/parsers/ast/mod.rs

```diff
@@ -28,65 +26,28 @@ type LocalProcMap = BTreeMap<String, (u16, ProcedureAst)>;
 /// A program has to have a body and no exported procedures.
 #[derive(Debug, Eq, PartialEq)]
 pub struct ProgramAst {
-    pub procedures: LocalProcMap,
+    pub local_procs: Vec<ProcedureAst>,
     pub body: Vec<Node>,
 }
 
-impl ProgramAst {
-    /// Returns byte representation of the `ProgramAst`.
-    pub fn to_bytes(&self) -> Vec<u8> {
-        let mut byte_writer = ByteWriter::new();
-
-        // procedures
-        byte_writer.write_u16(self.procedures.len() as u16);
-
-        for (_, proc) in sort_procs_by_index(&self.procedures) {
-            proc.write_into(&mut byte_writer);
-        }
-
-        // body
-        self.body.write_into(&mut byte_writer);
-
-        byte_writer.into_bytes()
-    }
-
-    /// Returns a `ProgramAst` struct by its byte representation.
-    pub fn from_bytes(bytes: &mut &[u8]) -> Result<Self, SerializationError> {
-        let mut byte_reader = ByteReader::new(bytes);
-
-        let mut procedures = LocalProcMap::new();
-
-        let num_procedures = byte_reader.read_u16()?;
-        for i in 0..num_procedures {
-            let proc_ast = ProcedureAst::read_from(&mut byte_reader)?;
-
-            procedures.insert(proc_ast.name.clone(), (i, proc_ast));
-        }
-
-        let body = Deserializable::read_from(&mut byte_reader)?;
-
-        Ok(ProgramAst { procedures, body })
-    }
-}
-
```

Comment:
I would keep these here. If we move them to tests now, we'll need to bring them back in the next PR.